tests/test.py

```diff
@@ -92,3 +92,48 @@ def test_adding_person_with_invalid_category(self):
         """Tests adding person who's category is not staff or fellow"""
         new = self.dojo.add_person("Wekesa','Maina", "Bootcamper", "N")
         self.assertEqual(new,"Wrong category. Can only be fellow or staff")
+
+class TestPrintingRoom(unittest.TestCase):
+    """ Test cases for printing Room and its occupants """
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def test_print_room_that_does_not_exist(self):
+        """ Should not print a room that is not in the system """
+        room_print = self.dojo.print_room('Entebbe')
+        self.assertEqual(room_print, "Room name Entebbe doesn't exist")
+
+    def test_prints_room_occupants(self):
+        """ Should print all the occupants of a specified room """
+        self.dojo.create_room("office", ["Django"])
+        self.dojo.add_person('Dennis', 'Wachiuri', 'Fellow')
+        room=self.dojo.print_room('Django')
+        self.assertIn("Dennis Wachiuri", "room")
+
+    def test_prints_empty_for_unoccupied_room(self):
+        """ Prints null if specified room is empty """
+        self.dojo.create_room("office", ["Django"])
+        r_print = self.dojo.print_room('Django')
+        self.assertEqual(r_print, "Django has no occupants")
+
+class TestPrintAllocatedUnallocated(unittest.TestCase):
+    """ Test case for printing allocations """
+    def setUp(self):
+        self.dojo = Dojo()
+
+    def test_prints_allocations_successfully(self):
+        """ should return allocated persons and unallocated persons"""
+        self.dojo.create_room("living", ["Django"])
+        # person 1 to 4 should be allocated to living Room Django
+        self.dojo.add_person('Dennis', 'Person1', 'Fellow')
+        self.dojo.add_person('Dennis', 'Person2', 'Fellow')
+        self.dojo.add_person('Dennis', 'Person3', 'Staff')
+        self.dojo.add_person('Dennis', 'Person4', 'Fellow')
+
```

Comment:
You forgot to include Y (yes) as the fourth parameter ....so they won't be allocated any livingspace
also staff should not be allocated living space